### ISSUE TITLE
"Branches are same" bug

### ISSUE THREAD

**Issue by 1itachi** [devops-infra/action-pull-request#51]
## :memo:  Brief description
Hi,

I am getting the message "Branches are the same" even though they are different.
I have a feature branch that is one commit ahead and a master which is also one commit ahead.

```
Comparing branches by diff...
fatal: remotes/origin/master...remotes/origin/automatic-sync: no merge base

[INFO] Both branches are the same. No action needed.
```
I believe this [line](https://github.com/devops-infra/action-pull-request/blob/master/entrypoint.sh#L66) is checking for a diff between the branches,  which will result in a 0 if both are ahead by one, instead, I think it should be checking for only feature branch.

**Comment by ChristophShyper** [comment]
Hi. I would need more information to investigate it and reproduce the issue.
Looks like a problem with git and checkout of the branches.
Are you using `actions/checkout` action? In which version and what are your parameters for it?

My action is running numerous times a day for many repos and it was used in all cases, with both branches being ahead of each other my few commits even without any issues.

**Comment by 1itachi** [comment]
Hi, thank you for getting back to me.
I am using `actions/checkout@v2` with only one parameter which is PAT. Should I be using the fetch-depth parameter? I did not have this issue about a week back.

**Comment by ChristophShyper** [comment]
Have you tried it with `actions/checkout@v2.3.4`?

**Comment by 1itachi** [comment]
I have not checked it with v2.3.4.
Here is the content:
```
name: sync template
on: 
  #schedule every 4 hours
  schedule:
   - cron: '0 */4 * * *'
   
jobs:
  check-version-change:
    runs-on: [ubuntu-latest]
    timeout-minutes: 30
    env:
       PR_BRANCH: automatic-sync #name of the PR branch
       FILE_PATTERN: :!template :!.actions #pattern or files to be committed
       PR_TITLE: "Apply changes from template" #default PR title

    #use personal token to checkout as GITHUB_TOKEN has limitation to update workflow file
    steps:
    - name: Stagger Start (25 min)
      uses: 1itachi/sleep-random@v1.0.0
      with:
        minutes: '25'
        random: 'true'
          
    - uses: actions/checkout@v2
      with:
        token: ${{ secrets.PAT }}
    
    - name: Extract branch name
      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
      id: extract_branch

    - name: Check branch exists
      run: |
          #Check if the branch exists in the remote
           existed_in_remote=$(git ls-remote --heads origin ${{ env.PR_BRANCH }})
           #If the branch is not present in remote
           if [ -z "$existed_in_remote" ]; then
              git push origin ${{ steps.extract_branch.outputs.branch }}:${{ env.PR_BRANCH }}
           fi 
    #check out a branch tracking remote branch (automated branch)
    - name: Checkout to remote branch
      run: |
            git fetch --all
            git checkout --track origin/${{ env.PR_BRANCH }}
    
    - name: Add files
      run: |
            **add files to the branch*****

     #commit to branch
    - name: Commit and push
      id: changes_found
      run: |
            git config --global user.email "action@github.com"
            git config --global user.name "GitHub Action"
            git add ${{ env.FILE_PATTERN }}
            git diff-index --quiet HEAD && echo "::set-output name=out::false" || echo "::set-output name=out::true"
            git diff-index --quiet HEAD || git commit -m "Apply changes from template" && git push origin ${{ env.PR_BRANCH }}
       
    - name: Create pull request
      if: ${{  steps.changes_found.outputs.out == 'true' }}
      uses: devops-infra/action-pull-request@master
      id: pr
      with:
          github_token: ${{ secrets.PAT }}
          source_branch: ${{ env.PR_BRANCH }}
          label: automated
          title: ${{ env.PR_TITLE }}
          reviewer: bcollamore
          body: "Auto-synced from template"

```

**Comment by ChristophShyper** [comment]
Are you sure the new branch is pushed properly to remote? 
Is the git user you are setting have proper permissions on the repository? 
Maybe missing `--all` for `git commit` is also causing it (depending on what is happening in the middle)?

Have you tried using https://github.com/devops-infra/action-commit-push instead?

Here you have example of how I'm using it to update my runner image on each dependency update: https://github.com/devops-infra/docker-terragrunt/blob/master/.github/workflows/CRON.yml

**Comment by 1itachi** [comment]
@ChristophShyper I tested this again but this time creating a branch manually on GitHub and making 1 commit of the feature so that it is ahead of master by 1 and making 1 commit on master so that it is ahead by 1. I experience the same error

**Comment by 1itachi** [comment]
@ChristophShyper Hi, I fixed the issue with adding `fetch-depth: 0` according to this https://stackoverflow.com/questions/64957915/github-actions-git-diff-master-head-fatal-no-merge-base

**Comment by ChristophShyper** [comment]
Yep, that would make sense. I use it always without any issues.

**Comment by ChristophShyper** [comment]
Everything works properly now?

**Comment by 1itachi** [comment]
Yes, I have not faced any issue after adding the `fetch depth`

**Comment by sanjeewaf** [comment]
i had this same issue even with fetch depth is 0.

How to reproduce.

1. create a feature branch from another branch ( I used: peterjgrainger/action-create-branch@v2.3.0)
2. Now Checkout a feature branch using the checkout action v 3 ( even with depth 0)
3.  Now change something
4. commit and push using stefanzweifel/git-auto-commit-action@v4
5.  Now try to use this action to create PR from the feature branch to another branch ( eg: develop)
you get the above error:  `no merge base`

fix I made is  after '4' above 
```
- name: Refresh Git
   run: git fetch --depth 1000 && git pull
```
